shared/settings/advanced/index.js

```diff
@@ -39,36 +41,29 @@ const Advanced = (props: Props) => (
   </Box>
 )
 
-type TraceButtonProps = {
-  durationSeconds: number,
-  traceInProgress: boolean,
-  onTrace: (durationSeconds: number) => void,
+type StartButtonProps = {
+  label: string,
+  inProgress: boolean,
+  onStart: () => void,
 }
 
-class TraceButton extends React.Component<TraceButtonProps> {
-  _onClick = () => {
-    this.props.onTrace(this.props.durationSeconds)
-  }
-
-  render() {
-    const label = `Trace (${this.props.durationSeconds}s)`
-    return (
-      <Button
-        waiting={this.props.traceInProgress}
-        style={{marginTop: globalMargins.small}}
-        type="Danger"
-        label={label}
-        onClick={this._onClick}
-      />
-    )
-  }
-}
+const StartButton = (props: StartButtonProps) => (
+  <Button
+    waiting={props.inProgress}
+    style={{marginTop: globalMargins.small}}
+    type="Danger"
+    label={props.label}
+    onClick={props.onStart}
+  />
+)
 
 type DeveloperState = {
   clickCount: number,
 }
 
 const clickThreshold = 7
+const traceDurationSeconds = 30
```

Comment:
A little weird to have the view care about how long this takes, but I guess it's nice to be able to reuse the action in a different place with a different duration. Could these consts (not `clickThreshold`) move into container though? then you could pass noarg functions